api/src/main/java/com/netflix/iceberg/encryption/PhysicalEncryptionKey.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.netflix.iceberg.encryption;
+
+import java.nio.ByteBuffer;
+
+/**
+ * The encryption key used to read and write encrypted files.
+ * <p>
+ * This key is never persisted in Iceberg, and is used in Iceberg clients (e.g. Spark, Pig, Hive)
+ * to encrypt and decrypt table data files.
+ */
+public interface PhysicalEncryptionKey {
+
+  /**
+   * The secret key material used for decrypting the data file.
+   */
+  ByteBuffer secretKeyBytes();
+
+  /**
+   * Key algorithm to use to convert the secret key bytes into a cryptographic KeyMaterial.
+   */
+  String keyAlgorithm();
+
+  /**
+   * The initialization vector bytes to use with this encryption key when generating ciphers.
+   */
+  ByteBuffer iv();
```

Comment:
An older version of the proposal proposed the idea of putting the iv in the key metadata. In reality the key header should be as generic and as opaque as possible. So we tie all aspects of the key material to the physical encryption key.

Reply:
Should this be optional? Not all algorithms use an IV. Also, is this something that should be stored with the data? The Parquet encryption spec stores the IV before each encrypted buffer.

Reply:
Also I modeled the way keys are stored after the way hadoop-crypto models encryption, which puts the iv inside the KeyMaterial object instead of inside the data - see https://github.com/palantir/hadoop-crypto/blob/ac8d4474ee667121873bf0abf0674d83c78d8b90/crypto-keys/src/main/java/com/palantir/crypto2/keys/KeyMaterial.java#L34

Reply:
> how opinionated does Iceberg want to be about what encryption algorithms it supports?

Iceberg should not be opinionated at all. That's why I keep pushing toward opaque values where an implementation can put whatever it needs to.